libjps/include/jps.h

```diff
@@ -0,0 +1,101 @@
+#ifndef _JPS_H_
+#define _JPS_H_
+
+#include <stdbool.h>
+#include <stdlib.h>
+
+typedef struct {
+    void * sim_context;
+} SimulationContext;
+
+typedef struct {
```

Comment:
So after each iteration this struct is filled with errors? Why not allow to setup optional function callbacks for that?

GLFW does this for example very nicely: https://www.glfw.org/docs/3.0/group__error.html